### PR TITLE
New subworkflow for scrna-seq data with lineage inference

### PR DESCRIPTION
**Pull request by saimukund20**

### CODE REVIEW

File: definitions/subworkflows/single_cell_rnaseq.cwl [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: Workflow
+label: "Running cellranger count and lineage"
```
Comment (chrisamiller) [inline definitions/subworkflows/single_cell_rnaseq.cwl]: s/lineage/lineage inference/

File: definitions/subworkflows/single_cell_rnaseq.cwl [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+inputs:
+    chemistry:
+        type: string?
+    reference:
+        type: Directory
+    sample_name:
+        type: string
+    fastq_directory:
+        type: Directory[]
+    lineage_min_cell:
```
Comment (chrisamiller) [inline definitions/subworkflows/single_cell_rnaseq.cwl]: s/cell/cells/

File: definitions/tools/cellmatch_lineage.cwl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        type: int?
+        default: 3
+        inputBinding:
+            position: 1 
+    lineage_min_features:
+        type: int?
+        default: 10
+        inputBinding:
+            position: 2
+
```
Comment (chrisamiller) [inline definitions/tools/cellmatch_lineage.cwl]: (Already talked with you about adding the lineage reference data as an input)

**Comment by chrisamiller** [comment]
looks really close!

File: definitions/tools/cellmatch_lineage.cwl [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+cwlVersion: v1.0
+class: CommandLineTool
+label: "Running a script to identify lineage of cells"
+
+baseCommand: ["/usr/local/bin/Rscript", "/opt/CellMatch_Haemopedia.r"]
+
+requirements:
+    - class: DockerRequirement
+      dockerPull: "saimukund/seurat_docker"
```
Comment (chrisamiller) [inline definitions/tools/cellmatch_lineage.cwl]: Adding a version tag to this would be good

Comment (tmooney) [inline definitions/tools/cellmatch_lineage.cwl]: It'd be good to have an explicit version for docker images used in our CWL.  (As a side note, we usually avoid including the word `docker` in the name of docker images.)

Comment (saimukund20) [reply definitions/tools/cellmatch_lineage.cwl]: Did the changes

Comment (saimukund20) [reply definitions/tools/cellmatch_lineage.cwl]: Changes made!

**Pull request merged**